File: DesignPatternsExercises/java/StateDesignPattern/Exercise/AtmMachine.java
public class AtmMachine implements AtmMachineStates {
    private final AtmMachineStates cardEjectedState = __________(1)__________();
    private final AtmMachineStates cardInsertedState = new CardInsertedState();
    private AtmMachineStates atmMachineState;

    public AtmMachine() {
        __________(2)__________ = cardEjectedState;
    }

    public String __________(3)__________() {
        return atmMachineState.getClass().getName();
    }

    @Override
    public boolean ejectCard() {
        if (__________(4)__________()) {
            atmMachineState = cardEjectedState;
            return true;
        }
        return false;
    }

    @Override
    public boolean insertCard() {
        if (__________(5)__________()) {
            atmMachineState = cardInsertedState;
            return true;
        }
        return false;
    }

    @Override
    public boolean withdraw() {
        return __________(6)__________();
    }
}

File: DesignPatternsExercises/ruby/state_design_pattern/exercise/atm_machine.rb
require_relative 'atm_machine_states'
require_relative 'card_ejected_state'
require_relative 'card_inserted_state'

class AtmMachine < AtmMachineStates
  def __________(1)__________
    @card_ejected_state = __________(2)__________
    @card_inserted_state = CardInsertedState.new
    __________(3)__________ = @card_ejected_state
  end

  def __________(4)__________
    @atm_machine_state.class
  end

  def eject_card?
    if __________(5)__________
      @atm_machine_state = @card_ejected_state
      return true
    end
    false
  end

  def insert_card?
    if __________(6)__________
      @atm_machine_state = @card_inserted_state
      return true
    end
    false
  end

  def withdraw?
    __________(7)__________
  end
end
